Tizen .NET Hot Reload feature update. (#1791)
Samsung/tizen-docs: docs/application/vstools/tools/dotnet-hotreload.md

# Tizen .NET Hot Reload

With Hot Reload you can now modify your apps managed source code while the application is being debugged, without the need to stop and restart the debugging.
After starting the debugging of the application on the emulator or real device, you can apply the changed file without stopping the debugging of the application.

> With Hot Reload changes are applied only on the second entrance to the function. If the change you made is supported your app will now be patched while its running with your new logic and you should see the changes in your app’s behavior the next time the updated code is re-executed by either your action or by something like a timer triggering the code.

Reviewer: ```suggestion
With Hot Reload changes are applied only on the second entrance to the function. If the change you made is supported, your app will now be patched while it's running with your new logic. You should see the changes in your app’s behavior the next time the updated code is re-executed by either your action or by a timer triggering the code.
```

## Enable Tizen .NET Hot Reload

Go to **Tools** menu in Visual Studio, and select **Tools > Options...** from the dropdown menu. **Options** window will open.

**1.** Go to **Tizen &gt; Tools** and check the **Enable Hot Reload** checkbox.

**Figure: Tizen options page to enable .NET Hot Reload**

![Tizen options page to enable .NET Hot Reload](media/enable_tizen_hotreload.png)

**2.** Go to **Debugging &gt; .NET / C++ Hot Reload** and check all the checkboxes.

**Figure: VS options page to enable .NET Hot Reload**

![Tizen options page to enable .NET Hot Reload](media/enable_vs_hotreload.png)

## Supported/Unsupported changes and architecture
Currently, Tizen .NET Hot Reload is supported on aarch64 and armv7l.

> No matter how you use .NET Hot Reload please be aware that some changes are not supported at runtime and will prompt you with a rude edit dialog and require you to restart your app in order to apply the changes. [Supported Edits in Tizen .NET Hot Reload](https://github.com/dotnet/roslyn/blob/main/docs/wiki/EnC-Supported-Edits.md)

Reviewer: External links in a .md file open in a new window
```suggestion
No matter how you use .NET Hot Reload please be aware that some changes are not supported at runtime and will prompt you with a rude edit dialog and require you to restart your app in order to apply the changes. [Supported Edits in Tizen .NET Hot Reload](https://github.com/dotnet/roslyn/blob/main/docs/wiki/EnC-Supported-Edits.md){:target="_blank"}
```

Reviewer 2: @omar-srbd Hi, could you review it again ?

Reviewer: @om90prakash, as per the commits history in this PR, I can see that this specific change was not made in previous commits.
The instance where the change was made and had to be reverted was in the video links of **lines 39 & 49**.
The change requested here is for an external link redirecting users to a different site and hence should open in a new tab.

# Examples

## Tizen .NET Hot Reload without breakpoint

This example shows using Hot Reload in an Tizen ElmSharp based UI application.

> Here, we are adding a callback method on button click event and once application is launched on target device and running in debug mode, we will modify the
> callback method and do Ctrl+S (Save) to apply Hot Reload.

Reviewer: ```suggestion
Here, we add a callback method on the button click event and once the application is launched on a target device and it is running in debug mode, we modify the callback method and Save (Ctrl+S) to apply Hot Reload.
```

> The following video demonstrates the same on Tizen Device:

Reviewer: ```suggestion
The following video demonstrates the example on a Tizen device:
```

> [![Hotreload Example Video](media/hotreload_without_bp.png)](media/Hotreload_Without_Breakpoint.mp4 "Hotreload Example Video Without Breakpoint")

Reviewer: ```suggestion
[![Hotreload Example Video](media/hotreload_without_bp.png)](media/Hotreload_Without_Breakpoint.mp4 "Hot Reload example video without breakpoint"){:target="_blank"}
```

> ## Tizen .NET Hot Reload with breakpoint

Reviewer: ```suggestion
### Tizen .NET Hot Reload with breakpoint
```

> This example shows using Hot Reload in an Tizen ElmSharp based UI application.

Reviewer: ```suggestion
The example shown here uses Hot Reload in a Tizen ElmSharp based UI application.
```

Here, we are adding a breakpoint in the main method on a static method call and once application is launched on target device and running in debug mode, we will modify the method and do ctrl+S (Save) to apply Hot Reload.

> The following video demonstrates the same on Tizen Device:

Reviewer: ```suggestion
The following video demonstrates the example on a Tizen device:
```

> [![Hotreload Example Video](media/hotreload_with_bp.png)](media/Hotreload_With_Breakpoint.mp4 "Hotreload Example Video With Breakpoint")

Reviewer: ```suggestion
[![Hotreload Example Video](media/hotreload_with_bp.png)](media/Hotreload_With_Breakpoint.mp4 "Hot Reload Example video with breakpoint"){:target="_blank"}
```
